cmake/common.cmake

```diff
@@ -245,14 +245,11 @@ if(SAN
    OR TSAN
    OR NOT USE_SNMALLOC
 )
-  set(SNMALLOC_LIB)
+  set(SNMALLOC_COMPILE_OPTIONS "")
 else()
-  set(SNMALLOC_ONLY_HEADER_LIBRARY ON)
-  # Remove the following two lines once we upgrade to snmalloc 0.5.4
-  set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
-  set(USE_POSIX_COMMIT_CHECKS off)
+  set(SNMALLOC_HEADER_ONLY_LIBRARY ON)
```

Comment:
This flag used to be set incorrectly (`SNMALLOC_ONLY_HEADER_LIBRARY` rather than `SNMALLOC_HEADER_ONLY_LIBRARY`). This is now fixed so we no longer generate `snmalloc_lib`.